test/unittest/tests/cursors/test_bulk_cursor.cpp

```diff
@@ -0,0 +1,379 @@
+            __wt_sleep(1, 0);
+            check_txn_updates("before commit", session_impl, diagnostics);
+
+            REQUIRE(session->commit_transaction(session, "") == expected_commit_result);
+            check_txn_updates("after commit", session_impl, diagnostics);
+        }
+    }
+}
+
+/*
+ * multiple_drop_test --
+ *     Ensure that a series of create/force drop operations on a table work as expected.
+ */
+static void
+multiple_drop_test(std::string const &config, int expected_open_cursor_result,
+  int expected_commit_result, bool do_sleep, bool diagnostics)
+{
+    ConnectionWrapper conn(DB_HOME);
+    std::string uri = "table:cursor_test";
+    std::string sleep_as_string = do_sleep ? "true" : "false";
+
+    SECTION("Multiple drop test: config = " + config + ", sleep = " + sleep_as_string)
+    {
+        int count = 0;
+        const int limit = 5;
+
+        while (count < limit) {
+            count++;
+
+            WT_SESSION_IMPL *session_impl = conn.createSession();
+            WT_SESSION *session = &session_impl->iface;
+
+            REQUIRE(session->create(session, uri.c_str(), "key_format=S,value_format=S") == 0);
+            REQUIRE(session->begin_transaction(session, "") == 0);
+
+            WT_CURSOR *cursor = nullptr;
+            int open_cursor_result =
+              session->open_cursor(session, uri.c_str(), nullptr, config.c_str(), &cursor);
+            REQUIRE(open_cursor_result == expected_open_cursor_result);
+
+            if (open_cursor_result == 0) {
+                insert_sample_values(cursor);
+
+                check_txn_updates("before close", session_impl, diagnostics);
+                REQUIRE(cursor->close(cursor) == 0);
+
+                if (diagnostics)
+                    printf("After close\n");
+
+                if (do_sleep)
+                    __wt_sleep(1, 0);
+            }
+
+            check_txn_updates("before drop", session_impl, diagnostics);
+            REQUIRE(session->drop(session, uri.c_str(), "force=true") == 0);
+
+            if (diagnostics)
+                printf("After drop\n");
+
+            if (do_sleep)
+                __wt_sleep(1, 0);
+
+            check_txn_updates("before checkpoint", session_impl, diagnostics);
+            REQUIRE(session->checkpoint(session, nullptr) == EINVAL);
+
+            if (do_sleep)
+                __wt_sleep(1, 0);
+
+            check_txn_updates("before commit", session_impl, diagnostics);
+            REQUIRE(session->commit_transaction(session, "") == expected_commit_result);
+            check_txn_updates("after commit", session_impl, diagnostics);
+            REQUIRE(session->close(session, nullptr) == 0);
+        }
+
+        // Confirm the correct number of loops were executed & we didn't exit early for any reason
+        REQUIRE(count == limit);
+    }
+}
+
+TEST_CASE("Cursor: bulk, non-bulk, checkpoint and drop combinations", "[cursor]")
+{
+    const bool diagnostics = false;
```

Comment:
I'm a fan of test cases - but this feels like something we'd have traditionally added as a Python test. Is there a reason it's more suitable as a unit test?

Reply:
The main benefit of developing these tests in C++ with Catch2 was that direct access from C++ to the C data structures and functions in WT made it much easier to diagnose the issues and thus focus the reproducer and narrow the problem down. 

While the resulting test could also be written in Python, it would have been harder to get to the point of identifying the exact test/reproducer steps required, and it would like have taken longer to diagnose the issue and actually develop the resulting test.

Reply:
Got it - this test seems useful, but I'd like to keep the Python testing complete as well. Could you also add a Python test now that you understand the ingredients necessary to trigger the problem?

Reply:
I'll have a go at creating one.

Reply:
I've added two Python tests (one for each of the two 'issues').